### DISCUSSION TITLE
How to set authentication whitelist?

### DISCUSSION THREAD

**Discussion by ghost** [honojs/hono#809] [Q&A] [answered]
Assuming that **most APIs require authentication**, but **only a few APIs do not require authentication**.

For example, `/login` and `/signup` need to skip authentication, but all other APIs need authentication.

Is there any **best practice** to solve this problem? Thanks!

**Comment by yusukebe** [comment, marked as answer]
How about doing this?:

```ts
const auth: MiddlewareHandler = async (_c, next) => {
  console.log('auth-middleware')
  await next()
}

const api = new Hono()
api.use('*', auth)
api.get('*', (c) => c.text('should be authorized!'))

const skip = new Hono()
skip.get('/api/*', (c) => c.text('skip!'))

const app = new Hono()

app.route('/', skip)
app.route('/', api)
```
